library/Icinga/Data/Db/DbQuery.php

```diff
@@ -222,7 +223,16 @@ protected function renderFilter($filter, $level = 0)
                 }
             }
         } else {
-            $str .= $this->whereToSql($filter->getColumn(), $filter->getSign(), $filter->getExpression());
+            if ($filter instanceof FilterEqual) {
+                $str .= sprintf(
+                    '%s %s %s',
+                    $filter->getColumn(),
+                    $filter->getSign(),
+                    $this->escapeForSql($filter->getExpression())
+                );
```

Comment:
This bypasses not only the wildcard rendering. There's also handling for arrays and the cases for `IS NULL` and `IS NOT NULL` which are lost now.

Since the method is deprecated anyway and (to be) removed by #4758, patch DbConnection::renderFilterExpression instead. There you can differentiate better as well.

Reply:
Changes made in DbConnection::renderFilterExpression, and the above changes are removed from DbQuery::renderFilter().